crates/symbolicator-service/src/services/symbolication/js.rs

```diff
@@ -110,37 +114,55 @@ async fn symbolicate_js_frame(
     lookup: &mut SourceMapLookup,
     raw_frame: &mut JsFrame,
     missing_sourcescontent: &mut u64,
-) -> Result<SymbolicatedJsFrame, JsFrameStatus> {
+) -> Result<JsFrame, JsModuleErrorKind> {
     let module = lookup.get_module(&raw_frame.abs_path).await;
 
     if !module.is_valid() {
-        return Err(JsFrameStatus::InvalidAbsPath);
+        return Err(JsModuleErrorKind::InvalidAbsPath);
     }
 
+    let mut frame = raw_frame.clone();
+
     // Apply source context to the raw frame
     apply_source_context_from_artifact(raw_frame, &module.minified_source);
 
     let smcache = match &module.smcache {
         Ok(smcache) => smcache,
-        Err(CacheError::Malformed(_)) => return Err(JsFrameStatus::MalformedSourcemap),
-        Err(_) => return Err(JsFrameStatus::MissingSourcemap),
-    };
-
-    let mut frame = SymbolicatedJsFrame {
-        status: JsFrameStatus::Symbolicated,
-        raw: raw_frame.clone(),
+        Err(CacheError::Malformed(_)) => return Err(JsModuleErrorKind::MalformedSourcemap),
+        Err(_) => return Err(JsModuleErrorKind::MissingSourcemap),
     };
 
     let (line, col) = match (raw_frame.lineno, raw_frame.colno) {
         (Some(line), Some(col)) if line > 0 && col > 0 => (line, col),
-        _ => return Err(JsFrameStatus::InvalidSourceMapLocation),
+        // Without a column we can still at least apply source context
```

Comment:
That already happens in `apply_source_context_from_artifact` above, though you moved the `frame = raw_frame.clone()` line above, so that the resolved source context is lost.

Just moving that line down would make this whole block unnecessary.